internal/manifest/custom_resource.go

```diff
@@ -45,16 +45,19 @@ func PostRunCreateCR(
 		return fmt.Errorf("failed to create resource: %w", err)
 	}
 
-	oMeta := &v1.PartialObjectMetadata{}
-	oMeta.SetName(obj.GetName())
-	oMeta.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
-	oMeta.SetNamespace(obj.GetNamespace())
-	oMeta.SetFinalizers(obj.GetFinalizers())
-	if added := controllerutil.AddFinalizer(oMeta, declarative.CustomResourceManager); added {
-		if err := kcp.Patch(
-			ctx, oMeta, client.Apply, client.ForceOwnership, client.FieldOwner(declarative.CustomResourceManager),
-		); err != nil {
-			return fmt.Errorf("failed to patch resource: %w", err)
+	// only add finalizer if Manifest is not under deletion
+	if obj.GetDeletionTimestamp().IsZero() {
+		oMeta := &v1.PartialObjectMetadata{}
```

Comment:
Can you extract this into a separate function where its name is explaining, could make the comment obsolete